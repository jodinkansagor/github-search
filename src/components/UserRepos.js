import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { getFetchedRepoInfo } from '../selectors/repoSelector';
import Repo from './Repo';
import styles from './Repos.css';


const UserRepos = () => {
  const userRepos = useSelector(getFetchedRepoInfo);

  const reposElements = userRepos.map(repo => (
    <li className={styles.repoListItem} key={repo.id}>
      <Repo {...repo} />
    </li>
  ));

  return (
    <section className={styles.repos}>
      <h1>Repositories</h1>
      <ul>
        {reposElements}
      </ul>
    </section>
  );

};

UserRepos.propTypes = {
  repo: PropTypes.any
};

export default UserRepos;
