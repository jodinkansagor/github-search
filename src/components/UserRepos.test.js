import React from 'react';
import { shallow } from 'enzyme';
import UserRepos from './UserRepos';
import { Provider } from 'react-redux';
import store from '../store';

describe('UserRepos component', () => {
  it('renders UserRepos', () => {
    const wrapper = shallow(<Provider store={store}><UserRepos /></Provider>).dive();
    expect(wrapper).toMatchSnapshot();
  });
});