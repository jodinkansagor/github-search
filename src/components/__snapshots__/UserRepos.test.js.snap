// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepos component renders UserRepos 1`] = `<UserRepos />`;
